ddtrace/contrib/internal/avro/patch.py

```diff
@@ -0,0 +1,74 @@
+import avro
+import wrapt
+
+from ddtrace import config
+from ddtrace.internal.utils.wrappers import unwrap
+from ddtrace.pin import Pin
+
+from .schema_iterator import SchemaExtractor
+
+
+config._add(
+    "avro",
+    dict(),
+)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(avro, "__version__", "")
+
+
+def patch():
+    """Patch the instrumented methods"""
+    if getattr(avro, "_datadog_patch", False):
+        return
+    avro._datadog_patch = True
+
+    _w = wrapt.wrap_function_wrapper
+
+    _w("avro.io", "DatumReader.read", _traced_deserialize)
+    _w("avro.io", "DatumWriter.write", _traced_serialize)
+    Pin(service=None).onto(avro.io.DatumReader)
+    Pin(service=None).onto(avro.io.DatumWriter)
+
+
+def unpatch():
+    if getattr(avro, "_datadog_patch", False):
+        avro._datadog_patch = False
+
+        unwrap(avro.io.DatumReader, "read")
+        unwrap(avro.io.DatumWriter, "write")
+
+
+#
+# tracing functions
+#
+def _traced_serialize(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
```

Comment:
I don't know what is the performance overhead of these lines, but all of this could be in the `if config._data_streams_enabled`

Reply:
fixed